pyccel/parser/semantic.py

```diff
@@ -3511,17 +3514,18 @@ def _visit_MacroFunction(self, expr, **settings):
         args = [a if isinstance(a, FunctionDefArgument) else FunctionDefArgument(a) for a in expr.arguments]
 
         def get_arg(func_arg, master_arg):
-            if isinstance(master_arg, FunctionDefArgument):
-                return FunctionDefArgument(func_arg.var.clone(master_arg.name), value = master_arg.default)
+            if isinstance(master_arg, PyccelSymbol):
+                return FunctionCallArgument(func_arg.var.clone(str(master_arg)))
             else:
-                return FunctionDefArgument(func_arg.var.clone(str(master_arg)))
+                return FunctionCallArgument(master_arg)
 
-        master_args = [get_arg(a,m) if isinstance(m, PyccelSymbol) else m
-                        for a,m in zip(func.arguments, expr.master_arguments)]
+        master_args = [get_arg(a,m) for a,m in zip(func.arguments, expr.master_arguments)]
 
         results = expr.results
-        macro   = MacroFunction(name, args, func, master_args,
-                                  results=results)
+        results_shapes = expr.results_shapes
```

Comment:
Are these two lines needed?

Can't you just pass `expr.results` and `expr.results_shapes` to `MacroFunction` below?

Reply:
Fixed 6701afb